homeassistant/components/shelly/switch.py

```diff
@@ -0,0 +1,165 @@
+"""Representation of Z-Wave switches."""
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.const import CONF_DEVICE_ID
+from homeassistant.core import callback
+
+from .const import (
+    COMMAND_SUFFIX,
+    CONF_MODEL,
+    CONF_TOPIC,
+    DOMAIN,
+    MODEL_SWITCHES,
+    MODELS,
+)
+
+ON = "on"
+OFF = "off"
+TOGGLE = "toggle"
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up Shelly switch(es) from config entry."""
+
+    device_id = config_entry.data[CONF_DEVICE_ID]
+    topic = config_entry.data[CONF_TOPIC]
+    model = config_entry.data[CONF_MODEL]
+    switches_count = MODELS[model][MODEL_SWITCHES]
+    switches = []
+    for i in range(0, switches_count):
+        switch_topic = topic + f"relay/{i}"
+        if i == 0:
+            unique_id = device_id
+            via_device = None
+        else:
+            unique_id = f"{device_id}_{i+1}"
+            via_device = device_id
+        switches.append(
+            ShellySwitch(
+                hass.components.mqtt, unique_id, switch_topic, model, via_device
+            )
+        )
+
+    async_add_entities(switches)
```

Comment:
Ideally I would simply create regular MQTT switches at this point (passing in the relevant info so that devices are created), rather than implementing the whole switch platform, but AFAIK using platforms from a different integration is against guidelines.

Reply:
Can you pin point that guideline ? maybe we can open an architecture issue to change that... this is just re-inventing the wheel....

Reply:
Unfortunately I have no idea how to find it, but I've seen several PRs that imported platforms from other components and core devs requested to remove that.
If I could rely on MQTT platforms it will greatly simplify both this PR and subsequent ones (as I intend to add sensor, binary sensor, and cover, at the least).

Reply:
BTW the new OZW integration, which I used for inspiration here, is also taking the same approach of implementing all of the platforms rather than relying on the existing MQTT platforms.